skrub/_dataframe/_common.py

```diff
@@ -326,7 +326,7 @@ def _concat_horizontal_pandas(*dataframes):
     init_index = dataframes[0].index
     dataframes = [df.reset_index(drop=True) for df in dataframes]
     dataframes = _join_utils.make_column_names_unique(*dataframes)
-    result = pd.concat(dataframes, axis=1, copy=False)
+    result = pd.concat(dataframes, axis=1)
```

Comment:
This should remove the deprecation warning. However, I'm wondering if we should opt-in for the new behaviour from now on using `pd.options.mode.copy_on_write = True` if available.

Reply:
Interesting! In which version was this feature introduced in pandas?

Reply:
It will be introduced in pandas 3.0 that is the next version. That's why the nightly build are catching this change of behaviour. However, I think that it could be cool to opt-in this behaviour to not have surprised between versions.

Reply:
So let's do via the pandas version then.